elements/lisk-transaction-pool/src/transaction_pool.ts

```diff
@@ -560,31 +560,35 @@ export class TransactionPool extends EventEmitter {
 			queueCheckers.checkTransactionForId(failedTransactions),
 		);
 
-		// Keep transactions in the ready queue which still exist
-		this._queues.ready.enqueueMany(
-			this._queues.ready.removeFor(
-				queueCheckers.checkTransactionForId(passedTransactions),
-			),
-		);
+		// Empty failed transactions list gurantees that all transactions can be processed together
+		// So move all passed transactions to the ready queue
+		if (failedTransactions.length === 0) {
```

Comment:
I think in case of
```
@verified queue
1. Valid Tx
2. Invalid Tx
```
failedTransactions length will be 1, and it will not move to ready queue?

Reply:
No, it will not be moved. I think if we optimize for these scenarios we will do it at the cost of complicating the logic. Complications will arise due to the fact that a) we'll be removing valid transactions from the verified queue and the pending queue and b) valid transactions will be only removed until we encounter first invalid transaction c) the function will get overcomplicated after dynamic fees because we reorder transactions from ready, pending and verified queues before processing them together.
This function is executed every second, and this scenario is a very narrow edge case. Hence, optimizing it does add many benefits for me. Do you disagree?